### ISSUE TITLE
Support for Relay

### ISSUE THREAD

**Issue by tillkruss** [shivammathur/setup-php#682]
**Describe the feature**

I'm one of the makers of Relay and we're heavy users `shivammathur/setup-php` ourselves and would love to see Relay support.

**Underlying issue**

Installing Relay is a little verbose on GitHub Actions, since it requires the `msgpack` and `igbinary` extensions and some extra steps.

**Describe alternatives**

```yaml
  - name: Setup PHP
    uses: shivammathur/setup-php@v2
    with:
      extensions: msgpack, igbinary

  - name: Install Relay
    run: |
      curl -L "https://builds.r2.relay.so/${{ env.relay }}/relay-${{ env.relay }}-php${{ env.php }}-debian-x86-64.tar.gz" | tar xz
      cd relay-${{ env.relay }}-php${{ env.php }}-debian-x86-64
      sudo cp relay.ini $(php-config --ini-dir)
      sudo cp relay-pkg.so $(php-config --extension-dir)/relay.so
      sudo sed -i "s/00000000-0000-0000-0000-000000000000/$(cat /proc/sys/kernel/random/uuid)/" $(php-config --extension-dir)/relay.so
```

**Additional context**

https://relay.so/docs/1.x/installation#github-actions

**Are you willing to submit a PR?**

Yes, but we don't know where to start. Happy to become sponsors as well 👍

**Comment by shivammathur** [comment]
@tillkruss 

Thanks for reaching out.
I will be happy to add support in the next release. Sponsorship is always helpful 👍

**Comment by tillkruss** [comment]
Excellent, happy to review ✌️

**Comment by shivammathur** [comment]
Added in 52eb86c60ca0879e41b29e94a354e87335260cb3. It is in develop right now, will be in the next release.

This should work on Linux and macOS once released.
```yml
- name: Install PHP
  uses: shivammathur/setup-php@v2
  with:
    php-version: '8.1' # 7.4 to 8.2
    extensions: relay # or specific version like relay-0.4.6 
  env:
    RELAY_KEY: ${{ secrets.RELAY_KEY }}
    RELAY_ENVIRONMENT: development
    RELAY_EVICTION_POLICY: lru
    RELAY_MAX_MEMORY: 256M
```

**Comment by tillkruss** [comment]
Amazing, looks good ❤️‍🔥

**Comment by ostrolucky** [comment]
Can we generate random key by default, in case no key was specified? Because if no key is specified, extension produces warnings. And I don't have an idea for a one-liner in RELAY_KEY to do this 🤔

**Comment by shivammathur** [comment]
@ostrolucky 

`RELAY_KEY` env would be for setting the license key.
https://relay.so/docs/1.x/configuration#default-configuration

The random uuid replacement in the extension binary will be done by default https://github.com/shivammathur/setup-php/blob/dd0115673c4a47a832dff844189234f0e9b3966e/src/scripts/extensions/relay.sh#L93-L98

**Comment by ostrolucky** [comment]
Ahh all good then, very nice

**Comment by shivammathur** [comment]
Released in [2.24.0](https://github.com/shivammathur/setup-php/releases/tag/2.24.0)
